crates/store/src/state.rs

```diff
@@ -203,15 +203,21 @@ impl State {
         // block store. Thus, such block should be considered as block candidates, but not
         // finalized blocks. So we should check for the latest block when getting block from
         // the store.
-        let store = self.block_store.clone();
+        let store = Arc::clone(&self.block_store);
         let block_save_task =
             tokio::spawn(async move { store.save_block(block_num, &block_data).await });
 
-        // scope to read in-memory data, validate the request, and compute intermediary values
-        let (account_tree, chain_mmr, nullifier_tree, notes) = {
+        // scope to read in-memory data, compute mutations required for updating account
+        // and nullifier trees, and validate the request
+        let (
+            nullifier_tree_old_root,
+            account_tree_old_root,
+            account_tree_update,
+            nullifier_tree_update,
+        ) = {
```

Comment:
nit: this arrangement looks a bit odd again. Maybe something like this will work better:
```rust
let (
    account_tree_old_root,
    account_tree_update,
    nullifier_tree_old_root,
    nullifier_tree_update,
) = {
```